src/UI/Components/Diagram/zoomHandler/zoomHandler.ts

```diff
@@ -0,0 +1,307 @@
+    if (!event.target) {
+      return;
+    }
+    const slider = event.target as HTMLInputElement;
+
+    this.updateSliderProgressBar(slider);
+  }
+
+  /**
+   * Updates the progress bar of the zoom slider.
+   *
+   * This method is responsible for updating the background of the zoom slider to reflect the current zoom level as currently chromium based browsers don't support styling of the progressbar.
+   * The zoom level is calculated as a percentage of the slider's current value relative to its minimum and maximum values.
+   *
+   * @param {HTMLInputElement} slider - The zoom slider, which is an HTML input element.
+   */
+  updateSliderProgressBar(slider: HTMLInputElement) {
+    const value =
+      ((Number(slider.value) - Number(slider.min)) /
+        (Number(slider.max) - Number(slider.min))) *
+      100;
+
+    slider.style.setProperty(
+      "--slider-background",
+      "linear-gradient(to right, var(--pf-v5-global--active-color--100) 0%, var(--pf-v5-global--active-color--100) " +
+        value +
+        "%, var(--pf-v5-global--palette--black-400) " +
+        value +
+        "%, var(--pf-v5-global--palette--black-400) 100%)",
+    );
+  }
+
+  /**
+   * Toggles the fullscreen mode of the document.
+   *
+   * This method is responsible for switching the document between fullscreen mode and normal mode.
+   */
+  toggleFullscreen() {
+    if (!document.fullscreenElement) {
+      document.documentElement.requestFullscreen();
+    } else if (document.exitFullscreen) {
+      document.exitFullscreen();
+    }
+  }
+
+  /**
+   * Changes the display style of an HTML element.
+   *
+   * This function is responsible for changing the display style of an HTML element.
+   * It uses the document.querySelector method to find the element and changes its display style.
+   *
+   * @param {string} selector - The CSS selector of the element to change.
+   * @param {string} display - The new display style for the element.
+   */
+  changeDisplay(selector: string, display: string) {
+    const element: HTMLElement | null = document.querySelector(selector);
+
+    if (element) {
+      element.style.display = display;
+    }
+  }
+
+  /**
+   * Updates the state of the toolbar buttons.
+   *
+   * This method is responsible for toggling visual state of the button responsible for toggling full screen mode,
+   * as well as the display of the page components when full screen mode is toggled.
+   *
+   * @method
+   */
+  updateFullscreenStyling() {
+    const fullscreenButton = this.toolbar.getWidgetByName(
+      "fullscreen",
+    ) as IconButton;
+
+    const canvas = document.querySelector("#canvas-wrapper");
+    const banners = document.querySelectorAll(".pf-v5-c-banner");
+
+    if (canvas) {
+      canvas.classList.toggle("fullscreen", !!document.fullscreenElement);
+    }
+
+    if (banners) {
+      banners.forEach(
+        (el) =>
+          ((el as HTMLElement).style.display = document.fullscreenElement
```

Comment:
you could have done the typing (eventually casting if really needed)  in the previous line, it would be more readable

Reply:
odd, but okay